repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilterChain.java

```diff
@@ -143,9 +147,13 @@ private void splitRequestHeaders(HttpServletRequestWrapper request) {
      */
     private List<FilterContext> getFilterChainForRequest(String uri) {
         List<FilterContext> filters = new LinkedList<>();
-        for (FilterContext filter : filterChainCopy) {
-            if (filter.getUriPattern() == null || filter.getUriPattern().matcher(uri).matches()) {
-                filters.add(filter);
+        if (bypassUrl.isPresent() && Pattern.compile(bypassUrl.get()).matcher(uri).matches()) {
```

Comment:
You could optionally replace the logical and operator here with more calls on the Optional.
`bypassUrl.map(url -> Pattern.compile(url).matcher(uri).matches()).orElse(false)`